crates/topos-tce/src/app_context/protocol.rs

```diff
@@ -83,10 +83,16 @@ impl AppContext {
                     error!("Unable to send Ready due to error: {e}");
                 }
             }
-
-            evt => {
-                debug!("Unhandled event: {:?}", evt);
+            ProtocolEvents::BroadcastFailed { certificate_id } => {
+                warn!("Broadcast failed for certificate {certificate_id}")
+            }
+            ProtocolEvents::AlreadyDelivered { certificate_id } => {
+                info!("Certificate {certificate_id} already delivered")
+            }
+            ProtocolEvents::Die => {
+                error!("The DoubleEcho unexpectedly died, this is unrecoverable")
```

Comment:
Should the process halt here? So `panic!` instead? Or…?

Reply:
Not really, The `Die` event is misleading I think, it's a leftover that I can need to be removed or refactor, but I hadn't time to dig deeper.